api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -492,7 +507,85 @@ private void attemptFirecloudWorkspaceCreation(FirecloudWorkspaceId workspaceId)
   @Override
   public ResponseEntity<CloneWorkspaceResponse> cloneWorkspace(String workspaceNamespace,
       String workspaceId, CloneWorkspaceRequest body) {
-    return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build();
+    Workspace workspace = body.getWorkspace();
+    if (workspace.getNamespace().equals("")) {
```

Comment:
This throws NPE if namespace isn't provided. Should be a similar "bad request" response.

Reply:
Good catch - fixed.

I was depending on the required attribute in swagger (as I assume `CreateWorkspace` was doing), but actually namespace isn't included for some reason, and it's a convoluted thing to depend on anyways.

Reply:
I get a 500 for this:
```
curl http://localhost:8081/api/v1/workspaces/test/test/clone -H "$AUTH_HEADER" -d '{"workspace": {"name": "", "namespace": "asdf", "researchPurpose": ""}}' -H 'Content-Type: application/json'
```
Error is:
```
Could not read JSON document: Can not construct instance of org.pmiops.workbench.model.ResearchPurpose: no String-argument constructor/factory method to deserialize from String value ('')
```
If you want to punt on this, I'm fine with that, but please file a bug if you do. In general, it's important to me that our API error messages are reasonable.

Reply:
Agreed this is bad, though it doesn't look specific to this clone endpoint (so I won't fix in this PR). Filed https://precisionmedicineinitiative.atlassian.net/browse/RW-319